src/inmanta/module.py

```diff
@@ -2202,7 +2202,10 @@ def load_module(
                 f"Could not find module {module_name}. Please make sure to add any module v2 requirements with"
                 " `inmanta module add --v2` and to install all the project's dependencies with `inmanta project install`."
             )
-
+        if isinstance(module, ModuleV1):
+            inmanta.warnings.warn(
+                f"Loaded V1 module {module.name}. The use of V1 modules is deprecated. Use the equivalent V2 module instead."
```

Comment:
We had a discussion about this last week, didn't we conclude all warnings raised via `inmanta.warnings.warn` should inherit from `InmantaWarning`? We could consider patching the method as follows:
```diff
diff --git a/src/inmanta/warnings.py b/src/inmanta/warnings.py
index 446f150d..3600cfde 100644
--- a/src/inmanta/warnings.py
+++ b/src/inmanta/warnings.py
@@ -168,8 +168,8 @@ class WarningsManager:
             logger.warning("%s", text)
 
 
-def warn(warning: InmantaWarning) -> None:
+def warn(warning: Union[str, InmantaWarning]) -> None:
     """
     Warn using the supplied InmantaWarning instance.
     """
-    warnings.warn(warning)
+    warnings.warn(warning if not isinstance(str) else InmantaWarning(warning))
```